halo2_proofs/src/poly/kzg/multiopen/shplonk/verifier.rs

```diff
@@ -62,7 +62,7 @@ where
         mut msm_accumulator: DualMSM<'params, E>,
     ) -> Result<Self::Guard, Error>
     where
-        I: IntoIterator<Item = VerifierQuery<'com, E::G1Affine, MSMKZG<E>>> + Clone,
+        I: IntoIterator<Item = VerifierQuery<'com, E::G1Affine, MSMKZG<'params, E>>> + Clone,
```

Comment:
Compiler complains here:

![image](https://github.com/taikoxyz/halo2/assets/22738317/124d58a2-d5d6-4c03-b9dc-5a9652192822)

but changing to 'com or adding a constraint `'params: 'com` then creates a not enough constrained lifetime.